lib/puppet/type/zfs_share.rb

```diff
@@ -0,0 +1,234 @@
+Puppet::Type.newtype(:zfs_share) do
```

Comment:
Generically speaking?  How do you feel about removing all the config properties and just replacing things like "sec_none_rw" with a config hash?  Maintaining all these properties looks tedious.  Could boil it down to just two properties, share_nfs and share_smb.